### ISSUE TITLE
Code Helper CPU Usage

### ISSUE THREAD

**Issue by jrob8577** [Microsoft/vscode#30740]
- VSCode Version: Code 1.14.1 (2648980a697a4c8fb5777dcfb2ab110cec8a2f58, 2017-07-13T19:08:57.197Z)
- OS Version: Darwin x64 16.6.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
material-icon-theme|PKi|2.0.10
atom-keybindings|ms-|2.9.1
---

Steps to Reproduce:

1. Open VS Code
2. Open a .ts or .js file

Reproduces without extensions: Yes

`ps aux` for offending process gives me the following:
```
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript/out/utils/electronForkStart /usr/local/lib/node_modules/typescript/lib/tsserver.js --useSingleInferredProject --enableTelemetry --cancellationPipeName /var/folders/fd/z5zndjm56t7d5xxkvb7wgy9r000sw7/T/vscode-tscancellation-56c26e143850f4f7072f.sock*
```

**Comment by jrob8577** [comment]
Note that I have installed typescript@2.3.4 as referenced in other issues, and have included the following in my user.settings:
```
    "files.exclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/CVS": true,
        "**/.DS_Store": true,
        "**/node_modules": true,
        "**/bower_components": true,
        "**/tmp": true
    },
    "files.watcherExclude": {
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true,
        "**/node_modules/**": true,
        "**/tmp/**": true,
        "**/bower_components/**": true
    },
    "typescript.tsdk": "/usr/local/lib/node_modules/typescript/lib"
```

**Comment by mjbvz** [comment]
Does this also happen if you use TypeScript 2.4.2? There were a few high cpu bugs they fixed in 2.4 that you could be hitting.

If you still see this issue in 2.4.2, can you please share the project or try collecting the tsserver logs:


1. Set `"typescript.tsserver.log": "verbose"`
1. Restart VSCode
1. Reproduce the issue
1. Run the `Open TSServer log File` command and view the `tsserver.log` file

If you can share the logs I can try looking through them to see what is going on

 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by jrob8577** [comment]
This looks like it may have been fixed with 2.4.2, I'll re-open if I can reproduce in the newer version.
